ot3_state_manager/ot3_state_manager/util.py

```diff
@@ -3,28 +3,81 @@
 from __future__ import annotations
 
 import enum
-import hashlib
+from enum import Enum, unique
 
+from opentrons.hardware_control.types import OT3Axis
 
-class Direction(enum.Enum):
+
+class Direction(int, enum.Enum):
     """Class representing direction to move axis in."""
 
-    POSITIVE = "+"
-    NEGATIVE = "-"
+    POSITIVE = 1
+    NEGATIVE = 0
 
     @classmethod
-    def from_symbol(cls, symbol: str) -> "Direction":
-        """Parse plus or minus symbol into direction."""
-        if symbol == "+":
-            return cls.POSITIVE
-        elif symbol == "-":
-            return cls.NEGATIVE
-        else:
+    def from_int(cls, direction_val: int) -> Direction:
+        """Parse integer into direction."""
+        if direction_val not in (Direction.NEGATIVE, Direction.POSITIVE):
+            raise ValueError(
+                "Value for direction must be either 0 (Negative) or 1 "
+                f"(Positive). You passed {direction_val}."
+            )
+        return Direction(direction_val)
+
+
+class SyncPinState(int, enum.Enum):
+    """Class representing sync pin state."""
+
+    HIGH = 1
+    LOW = 0
+
+    @classmethod
+    def from_int(cls, state_val: int) -> SyncPinState:
+        """Parse integer into sync pin state."""
+        if state_val not in (SyncPinState.LOW, SyncPinState.HIGH):
             raise ValueError(
-                f'Symbol "{symbol}" is not valid.' f' Valid symbols are "+" and "-"'
+                "Value for state must either be 0 (LOW) or 1 (HIGH). "
+                f"You passed {state_val}."
             )
+        return SyncPinState(state_val)
+
 
+@unique
+class MoveMessageHardware(Enum):
+    """Enum representing a mapping of an integer hardware id to an OT3Axis object."""
 
-def get_md5_hash(data: bytes) -> bytes:
-    """Convert bytes into a md5 hash."""
-    return hashlib.md5(data).hexdigest().encode()
+    def __init__(self, hw_id: int, axis: OT3Axis) -> None:
+        """Create a MoveMessageHardware object."""
+        self.hw_id = hw_id
+        self.axis = axis
+
+    # hw_id, axis
+    X_AXIS = 0x0000, OT3Axis.X
+    Y_AXIS = 0x0001, OT3Axis.Y
+    Z_L_AXIS = 0x0002, OT3Axis.Z_L
+    Z_R_AXIS = 0x0003, OT3Axis.Z_R
+    Z_G_AXIS = 0x0004, OT3Axis.Z_G
+    P_L_AXIS = 0x0005, OT3Axis.P_L
+    P_R_AXIS = 0x0006, OT3Axis.P_R
+    G_AXIS = 0x0007, OT3Axis.G
+    Q_AXIS = 0x0008, OT3Axis.Q
```

Comment:
What does Q axis represent here?

Reply:
To my understanding, the Q axis is the high-throughput pipette axis.
https://github.com/Opentrons/opentrons/blob/edge/api/src/opentrons/hardware_control/types.py#L128